query-engine/connector-test-kit-rs/query-engine-tests/tests/new/relation_load_strategy.rs

```diff
@@ -86,7 +86,9 @@ mod relation_load_strategy {
 
     async fn assert_used_lateral_join(runner: &mut Runner, expected: bool) {
         let logs = runner.get_logs().await;
-        let actual = logs.iter().any(|l| l.contains("LEFT JOIN LATERAL"));
+        let actual = logs
+            .iter()
+            .any(|l| l.contains("LEFT JOIN LATERAL") || (l.contains("JSON_ARRAYAGG") && l.contains("JSON_OBJECT")));
```

Comment:
Since we're not using JOINs to resolve relations on MySQL, this had to be changed with a loose and worse heuristic.

Reply:
I think it's fine. Should we maybe now rename the function to something else? Same problem as https://github.com/prisma/prisma-engines/pull/4639#discussion_r1478655075 I guess.